tests/unit_tests/test_lobbyconnection.py

```diff
@@ -255,3 +256,230 @@ def test_avatar_select_no_avatar(mock_query, fa_server_thread):
+
+
+    fa_server_thread.handleAction('UPLOAD_MAP', stream)
+    fa_server_thread.sendJSON.assert_called_once_with( \
+        dict(command="notice", style="info", text="Map correctly uploaded."))
+
+@mock.patch('src.lobbyconnection.Config')
+@mock.patch('src.lobbyconnection.QSqlQuery')
+@mock.patch('src.lobbyconnection.QFile')
+@mock.patch('src.lobbyconnection.zipfile')
+def test_handle_action_upload_map_invalid_zip(mock_zipfile, mock_qfile, mock_query, mock_config, fa_server_thread):
+    fa_server_thread.sendJSON = mock.Mock()
+
+    stream = mock.Mock()
+    zipMap = mock.MagicMock()
+    infos = json.dumps({'name': '', 'description': '', 'max_players': '',
+                        'map_type': '', 'battle_type': '', 'map_size': {'0': '', '1': ''},
+                        'version': ''})
+    stream.readQString.side_effect = ['', '', zipMap, infos, 0, mock.Mock()]
+
+    # fake no db entry exists
+    mock_query.return_value.size.return_value = 0
+
+    # is invalid zip
+    mock_zipfile.is_zipfile.return_value = False
+
+    fa_server_thread.handleAction('UPLOAD_MAP', stream)
+    fa_server_thread.sendJSON.assert_called_once_with( \
+        dict(command="notice", style="error", text="Cannot unzip map. Upload error ?"))
+
+@mock.patch('src.lobbyconnection.Config')
+@mock.patch('src.lobbyconnection.QSqlQuery')
+@mock.patch('src.lobbyconnection.QFile')
+@mock.patch('src.lobbyconnection.zipfile')
+def test_handle_action_upload_map_exists(mock_zipfile, mock_qfile, mock_query, mock_config, fa_server_thread):
+    fa_server_thread.sendJSON = mock.Mock()
+
+    stream = mock.Mock()
+    zipMap = mock.MagicMock()
+    infos = json.dumps({'name': '', 'description': '', 'max_players': '',
+                        'map_type': '', 'battle_type': '', 'map_size': {'0': '', '1': ''},
+                        'version': ''})
+    stream.readQString.side_effect = ['', '', zipMap, infos, 0, mock.Mock()]
+
+    # map allready exists
+    mock_query.return_value.size.return_value = 1
+
+    fa_server_thread.handleAction('UPLOAD_MAP', stream)
+    (response, ), _ = fa_server_thread.sendJSON.call_args
+    assert response['command'] == 'notice'
+    assert response['style'] == 'error'
+
+def test_handle_action_upload_map_invalid_messages(fa_server_thread):
+    fa_server_thread.sendJSON = mock.Mock()
+
+    zipMap = mock.MagicMock()
+    infos = {'name': '', 'description': '', 'max_players': '',
+             'map_type': '', 'battle_type': '', 'map_size': {'0': '', '1': ''},
+             'version': ''}
+
+    error_messages = {'name': 'No map name provided.',
+                      'description': 'No map description provided.',
+                      'max_players': 'No max players provided.',
+                      'map_type': 'No map type provided.',
+                      'battle_type': 'No battle type provided.',
+                      'map_size': 'No map size provided.',
+                      'version': 'No version provided.'}
```

Comment:
This only tests whether the message contains they keys, we should also verify each of them individually.